frontend/public/components/monitoring/alert-manager-yaml-editor.tsx

```diff
@@ -28,70 +27,59 @@ const AlertManagerYAMLEditor: React.FC<AlertManagerYAMLEditorProps> = ({
 
   const [errorMsg, setErrorMsg] = React.useState(initErrorMsg);
   const [successMsg, setSuccessMsg] = React.useState();
-  const [inProgress, setInProgress] = React.useState(false);
-  const [alertManagerYamlStr, setAlertManagerYamlStr] = React.useState(
-    !_.isEmpty(encodedAlertManagerYaml) ? Base64.decode(encodedAlertManagerYaml) : '',
-  );
+  const alertManagerYamlStr = !_.isEmpty(encodedAlertManagerYaml)
+    ? Base64.decode(encodedAlertManagerYaml)
+    : '';
 
-  const save = (e) => {
-    e.preventDefault();
-    if (_.isEmpty(alertManagerYamlStr)) {
+  const save = (yaml: string) => {
+    if (_.isEmpty(yaml)) {
       setErrorMsg('Alertmanager configuration cannot be empty.');
       setSuccessMsg('');
       return;
     }
-    setInProgress(true);
-    patchAlertManagerConfig(secret, alertManagerYamlStr).then(
+    patchAlertManagerConfig(secret, yaml).then(
       (newSecret) => {
         setSuccessMsg(
           `${newSecret.metadata.name} has been updated to version ${
             newSecret.metadata.resourceVersion
           }`,
         );
         setErrorMsg('');
-        setInProgress(false);
       },
       (err) => {
         setErrorMsg(err.message);
         setSuccessMsg('');
-        setInProgress(false);
       },
     );
   };
 
   return (
-    <div className="co-m-pane__body">
-      <form
-        className="co-m-pane__body-group co-alert-manager-yaml-form co-m-pane__form"
-        onSubmit={save}
-      >
-        <p className="co-alert-manager-yaml__explanation">
+    <>
+      <div className="co-alertmanager-yaml-editor__header">
```

Comment:
There are layout issues at narrow screen widths. I think you'd be better off using `co-m-nav-title` like.

```
<>
  <div className="co-m-nav-title">
    <p className="help-block">
      ...
    </p>
    {errorMsg && ...
    {successMsg && ...
  </div>
  <EditAlertmanagerYAML onSave={save} obj={alertManagerYamlStr} />
</>
```

Then `co-alertmanager-yaml-editor__header` and `co-alertmanager-yaml-alert` can be simplified or possibly removed.

Reply:
Replaced `co-alertmanager-yaml-editor__header` with `co-m-nav-title`.   
`co-alertmanager-yaml-alert` is used to align the inline alerts:
![image](https://user-images.githubusercontent.com/12733153/69967453-6d7a8100-14e6-11ea-8e6c-f22795599d5a.png)

Reply:
Looks better, but I'm still seeing layout issues at small screen widths. I think the solution is to put the `Alert`s inside the `co-m-nav-title`, which should handle the alignment automatically.

![screenshot](https://user-images.githubusercontent.com/460802/70021907-99c0fb00-15d5-11ea-9c03-551c1c37fb45.png)

Reply:
Hi @kyoto, thanks, putting alerts inside of `co-m-nav-title` did fix the left alignment, but still wasn't happy with large bottom gap between alerts and text editor.  Updated `co-alertmanager-yaml-alert` to display properly on **small screen**:
![image](https://user-images.githubusercontent.com/12733153/70158323-0ba44d80-1685-11ea-9c3b-389b9990f770.png)

**Md/Lg screen:**

![image](https://user-images.githubusercontent.com/12733153/70158449-3f7f7300-1685-11ea-9f54-ee7812f61b73.png)

- thanks

Reply:
UPDATE:  Passed in high level component's Alerts as children to YAML Editor.  Ie. Alerts now appear below textarea -no need for custom css.